vllm/entrypoints/openai/reasoning_parsers/deepseek_r1_reasoning_parser.py

```diff
@@ -124,25 +120,42 @@ def extract_reasoning_content_streaming(
     def extract_reasoning_content(
             self, model_output: str, request: ChatCompletionRequest
     ) -> tuple[Optional[str], Optional[str]]:
+        """
+        Extract reasoning content from the model output.
+
+        For text <think>abc</think>xyz:
+        - 'abc' goes to reasoning_content
+        - 'xyz' goes to content
+
+        Returns:
+            tuple[Optional[str], Optional[str]]: reasoning content and content
+        """
+
+        # Check if the start token is present in the model output, remove it
```

Comment:
There may be a bug about regex, according to the issue, but I have not reproduced it.

But I found a bug about old chat templates: `<think>This is a reasoning section` will result `reasoning_content: <think>This is a reasoning section`.